### DISCUSSION TITLE
rabbitmq-sharding plugin - how to delete sharding queues

### DISCUSSION THREAD

**Discussion by liu0013** [rabbitmq/rabbitmq-server#3004] [Other]
If I want to delete sharding queues manually (auto-delete=false) then is there any easier way to delete sharding queues programmatically?  I found the rabbitmq-sharding test cases only support delete sharding queue one by one based on node name - https://github.com/rabbitmq/rabbitmq-sharding/blob/master/test/src/rabbit_sharding_SUITE.erl#L319-L324 
is possible to delete sharding queues by logic queue name?

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
No, I don't think so. Queue shards are regular queues named using a convention. You can delete them one by one. If you don't know how many shards a node have you can overshoot as deleting a non-existent queue is idempotent (does not produce a channel exception) starting with 3.0.

**Reply by liu0013** [reply]
@michaelklishin thank you!  if rabbitmq-sharding plugin can support to delete batch sharding queues by logic queue name then it's convenient for rabbitmq client.
